mars/dataframe/groupby/fill.py

```diff
@@ -0,0 +1,178 @@
+            return result_df.dtypes
+
+    def __call__(self, groupby):
+        in_df = groupby
+        while in_df.op.output_types[0] not in (OutputType.dataframe, OutputType.series):
+            in_df = in_df.inputs[0]
+        out_dtypes = self._calc_out_dtypes(groupby)
+
+        kw = in_df.params.copy()
+        kw["index_value"] = parse_index(pd.RangeIndex(-1), groupby.key)
+        if self.output_types[0] == OutputType.dataframe:
+            kw.update(
+                dict(
+                    columns_value=parse_index(out_dtypes.index, store_data=True),
+                    dtypes=out_dtypes,
+                    shape=(groupby.shape[0], len(out_dtypes)),
+                )
+            )
+        return self.new_tileable([groupby], **kw)
+
+    @classmethod
+    def tile(cls, op):
+        in_groupby = op.inputs[0]
+        out_df = op.outputs[0]
+
+        chunks = []
+        for c in in_groupby.chunks:
+            new_op = op.copy().reset_key()
+
+            new_index = parse_index(pd.RangeIndex(-1), c.key)
+            if op.output_types[0] == OutputType.dataframe:
+                chunks.append(
+                    new_op.new_chunk(
+                        [c],
+                        index=c.index,
+                        shape=(np.nan, len(out_df.dtypes)),
+                        dtypes=out_df.dtypes,
+                        columns_value=out_df.columns_value,
+                        index_value=new_index,
+                    )
+                )
+        new_op = op.copy().reset_key()
+        kw = out_df.params.copy()
+        kw["chunks"] = chunks
+        if op.output_types[0] == OutputType.dataframe:
+            kw["nsplits"] = ((np.nan,) * len(chunks), (len(out_df.dtypes),))
+        return new_op.new_tileables([in_groupby], **kw)
+
+    @classmethod
+    def execute(cls, ctx, op: "GroupByFillOperand"):
+        in_data = ctx[op.inputs[0].key]
+        out_chunk = op.outputs[0]
+
+        if not in_data or in_data.empty:
+            ctx[out_chunk.key] = (
+                build_empty_df(out_chunk.dtypes)
+                if op.output_types[0] == OutputType.dataframe
+                else build_empty_series(out_chunk.dtype, name=out_chunk.name)
+            )
+            return
+
+        func_name = getattr(op, "_func_name")
+        if func_name == "fillna":
+            ctx[out_chunk.key] = in_data.fillna(
+                value=op.value,
+                method=op.method,
+                axis=op.axis,
+                limit=op.limit,
+                downcast=op.downcast,
+            )
+        else:
+            result = getattr(in_data, func_name)(limit=op.limit)
+            if result.ndim == 2:
+                ctx[out_chunk.key] = result.astype(out_chunk.dtypes, copy=False)
+            else:
+                ctx[out_chunk.key] = result.astype(out_chunk.dtype, copy=False)
+
+
+class GroupByFFill(GroupByFillOperand):
+    _op_type_ = opcodes.FILL_NA
```

Comment:
About `op_type`, is it better to add a new one called `GROUPBY_FILL_NA` or just use `FILL_NA`?

https://github.com/mars-project/mars/blob/1f6c3d4d7296fad475ce707bfa00104a1ec3edc7/mars/opcodes.py#L421

I prefer to add a new one. What's your opinion @wjsi ?

Reply:
Other GroupBy operands use

```
_op_type_ = opcodes.APPLY
_op_module_ = "dataframe.groupby"
```

Seems to be a better choice?